src/services/project-service/project.service.ts

```diff
@@ -0,0 +1,69 @@
+import {Injectable} from '@angular/core';
+import {Project, PROJECT_KEY} from '../../app/interfaces/project';
+import {LocalStorageService} from '../storage-service/local-storage-service.service';
+import {PopupService} from '../popup-service/popup.service';
+import {Observable} from 'rxjs';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProjectService {
+  constructor(private localStorageService: LocalStorageService,  private popUpService: PopupService) { }
+
+  private taskExists(task: string, currentTasks: Project[]): Project {
+    for (const project of currentTasks) {
+      if (task === project.name) {
+        return project;
+      }
+    }
+    return null;
+  }
+  public getProjects(): Observable<Project[]> {
+    return this.localStorageService.getStorage(PROJECT_KEY);
+  }
+  addProject(task: Project): void {
```

Comment:
What happens if the server has returned an error? How can you show notification when the project has been created?
These are the reasons we always return an observable from the service. This way, the component can react to the result. Also, subscribing requires unsubscribing, usually the component does both...